src/plugins/lvm-dbus.c

```diff
@@ -2445,6 +2446,41 @@ gboolean bd_lvm_lvresize (const gchar *vg_name, const gchar *lv_name, guint64 si
     return call_lv_method_sync (vg_name, lv_name, "Resize", params, NULL, extra, TRUE, error);
 }
 
+/**
+ * bd_lvm_lvrepair:
+ * @vg_name: name of the VG containing the to-be-repaired LV
+ * @lv_name: name of the to-be-repaired LV
+ * @pv_list: (array zero-terminated=1): list of PVs to be used for the repair
+ * @extra: (nullable) (array zero-terminated=1): extra options for the LV resize
+ *                                                 (just passed to LVM as is)
+ * @error: (out) (optional): place to store error (if any)
+ *
+ * Returns: whether the @vg_name/@lv_name LV was successfully repaired or not
+ *
+ * Tech category: %BD_LVM_TECH_BASIC-%BD_LVM_TECH_MODE_MODIFY
+ */
+gboolean bd_lvm_lvrepair (const gchar *vg_name, const gchar *lv_name, const gchar **pv_list, const BDExtraArg **extra, GError **error) {
```

Comment:
I still don't like the idea of using LVM cli in the DBus plugin. I know that it's not possible to have lvmdbusd without lvm so this isn't really adding a new dependency, but it still makes `bd_lvm_is_tech_avail` a little bit meaning less for the DBus plugin in this case. I would prefer not using the cli here and just making the function return an error, because LVM DBus doesn't (yet) support repair.

@tbzatek @vpodzime thoughts?

Reply:
I completely agree. That's why we have the flags and checks.

Reply:
Roger!

Reply:
It's not clear to me how this should be wired up, unfortunately... so I have just put a g_set_error in there. Could you explain how this should look properly?

Reply:
Just set error to something like

```
g_set_error (error, BD_LVM_ERROR, BD_LVM_ERROR_TECH_UNAVAIL, "lvrepair is not supported by this plugin implementation.");
```
and return `FALSE` from the function.